### PR TITLE
Remove dependence on Eris objects as params from helper methods 

### PR DESCRIPTION
**Pull request by Brainicism**

### CODE REVIEW

File: src/structures/message_context.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import Eris from "eris";
+import { getUserTag } from "../helpers/discord_utils";
+import state from "../kmq";
+import KmqMember from "./kmq_member";
+
+export default class MessageContext {
+    /** The text channel to send the message to */
+    public textChannelId: string;
```
Comment (taahamahdi) [inline src/structures/message_context.ts:8]: textChannelID

File: src/structures/message_context.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    /** The author to represent the message as */
+    public author: KmqMember;
+
+    /** The guild ID to send the message to */
+    public guildId: string;
```
Comment (taahamahdi) [inline src/structures/message_context.ts:14]: guildID

File: src/structures/game_session.ts [hunk 3]
```diff
@@ -61,16 +63,16 @@ export default class GameSession {
...
-    public readonly owner: Eris.User;
+    public readonly owner: KmqMember;
 
     /** The Scoreboard object keeping track of players and scoring */
     public readonly scoreboard: Scoreboard;
 
-    /** The Eris.TextChannel in which the GameSession was started in, and will be active in */
-    public readonly textChannel: Eris.TextChannel;
+    /** The ID of text channel in which the GameSession was started in, and will be active in */
+    public readonly textChannelId: string;
```
Comment (taahamahdi) [inline src/structures/game_session.ts:72]: textChannelID

File: src/structures/game_session.ts [hunk 4]
```diff
@@ -61,16 +63,16 @@ export default class GameSession {
...
 
-    /** The Eris.VoiceChannel in which the GameSession was started in, and will be active in */
-    public readonly voiceChannel: Eris.VoiceChannel;
+    /** The ID of the voice channel in which the GameSession was started in, and will be active in */
+    public readonly voiceChannelId: string;
```
Comment (taahamahdi) [inline src/structures/game_session.ts:75]: voiceChannelID

File: src/commands/game_commands/skip.ts [hunk 5]
```diff
@@ -15,30 +14,31 @@ import _logger from "../../logger";
 import GameRound from "../../structures/game_round";
 import { GuildTextableMessage } from "../../types";
 import { KmqImages } from "../../constants";
+import MessageContext from "../../structures/message_context";
 
 const logger = _logger("skip");
 
 function getSkipsRequired(message: GuildTextableMessage): number {
-    return Math.floor(getNumParticipants(message) * 0.5) + 1;
+    return Math.floor(getNumParticipants(message.member.voiceState.channelID) * 0.5) + 1;
```
Comment (taahamahdi) [inline src/commands/game_commands/skip.ts:22]: Can we do ceiling here instead of floor + 1?

File: src/commands/game_commands/begin.ts [hunk 6]
```diff
@@ -2,35 +2,42 @@ import BaseCommand, { CommandArgs } from "../base_command";
...
+import MessageContext from "../../structures/message_context";
+import GameSession from "../../structures/game_session";
 
 const logger = _logger("begin");
 
 export default class BeginCommand implements BaseCommand {
-    async call({ message, gameSessions }: CommandArgs) {
-        const { guildID, author } = message;
-        const gameSession = gameSessions[guildID];
+    canStart(gameSession: GameSession, authorId: string, messageContext: MessageContext): boolean {
```
Comment (taahamahdi) [inline src/commands/game_commands/begin.ts:14]: authorID

File: src/structures/kmq_member.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import Eris from "eris";
+import { getUserTag } from "../helpers/discord_utils";
+
+export default class KmqMember {
```
Comment (taahamahdi) [inline src/structures/kmq_member.ts:4]: Would it make sense to instead reuse player.ts here? A lot of the properties are shared, you could just move the new method over there

**Review by taahamahdi — CHANGES_REQUESTED** [review]

File: src/structures/kmq_member.ts (hunk 7 above)
Comment (Brainicism) [reply src/structures/kmq_member.ts:4]: We can look into making Player inherit this later

**Review by taahamahdi — APPROVED** [review]

**Pull request merged**